src/cast_core/src/KafkaLoader.cpp

```diff
@@ -0,0 +1,113 @@
+// Copyright 2023-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cast_core/actors/KafkaLoader.hpp>
+
+#include <memory>
+
+#include <yaml-cpp/yaml.h>
+
+#include <bsoncxx/json.hpp>
+
+#include <mongocxx/client.hpp>
+#include <mongocxx/collection.hpp>
+#include <mongocxx/database.hpp>
+
+#include <boost/log/trivial.hpp>
+#include <boost/throw_exception.hpp>
+
+#include <gennylib/Cast.hpp>
+#include <gennylib/MongoException.hpp>
+#include <gennylib/context.hpp>
+
+#include <value_generators/DocumentGenerator.hpp>
+
+namespace genny::actor {
+
+namespace {
+
+// Kafka producer flush timeout. Flush is called after each phase.
+static constexpr int kKafkaFlushTimeoutMs = 10'000;
+
+};
+
+struct KafkaLoader::PhaseConfig {
+    DocumentGenerator documentExpr;
+
+    PhaseConfig(PhaseContext& phaseContext, ActorId id) :
+        documentExpr{phaseContext["Document"].to<DocumentGenerator>(phaseContext, id)} {}
+};
+
+void KafkaLoader::run() {
+    for (auto&& config : _loop) {
+        for (const auto&& _ : config) {
+            auto document = config->documentExpr();
+            std::string json = bsoncxx::to_json(document.view());
+
+            auto inserts = _totalInserts.start();
+            BOOST_LOG_TRIVIAL(debug) << " KafkaLoader Inserting " << json;
```

Comment:
Logging might introduce unpredictable performance variance, so I think we should do any logging outside of the code path being measured as much as possible:

```suggestion
            BOOST_LOG_TRIVIAL(debug) << " KafkaLoader Inserting " << json;
            auto inserts = _totalInserts.start();
```